### PR TITLE
Avoid incorrect tuple transformation in single-element case (`C409`)

### PR DESCRIPTION
**Pull request by WindowGenerator**
# Summary
Fixed: incorrect rule transformation rule C409 with single element.
# Test Plan
Added examples from #10323 to test fixtures.

### CODE REVIEW

File: crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_literal_within_tuple_call.rs [hunk 1]
```diff
@@ -92,23 +95,26 @@ pub(crate) fn unnecessary_literal_within_tuple_call(checker: &mut Checker, call:
...
-        // Replace from the end of the inner list or tuple to the end of the call with `)`.
-        let call_end = Edit::replacement(
-            ")".to_string(),
-            argument.end() - TextSize::from(1),
-            call.end(),
-        );
+        // Replace [1, 2] -> (1, 2) or [1] -> (1,)
+        let tuple: String = if let [elt] = elts.as_slice() {
+            let elt: &str = checker.locator().slice(elt);
+            format!("({elt},)")
```
Comment (charliermarsh) [inline crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_literal_within_tuple_call.rs:109]: This is going to lose the spacing around the element. Can you take a look at how we solved this in https://github.com/astral-sh/ruff/pull/10461?

Comment (charliermarsh) [reply crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_literal_within_tuple_call.rs:109]: For example: https://github.com/astral-sh/ruff/pull/10461/files#diff-8c56901e841c4e8733b61b77c68b174ea49dc98d02221715acafc8cfff273971R734

Comment (WindowGenerator) [reply crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_literal_within_tuple_call.rs]: Thank you for suggestion!

**Review by charliermarsh — APPROVED** [review]
Thanks!

**Pull request merged**